src/Wellcome.Dds/Utils/Caching/BinaryObjectCache.cs

```diff
@@ -102,7 +102,8 @@ public T GetCachedObject(string key, Func<T> getFromSource, Predicate<T> storedV
                         {
                             logger.LogDebug("Cache MISS for {0}, will attempt read from disk", memoryCacheKey);
                         }
-                        t = storage.Read<T>(cachedFile);
+                        // cannot await in the body of a lock statement
```

Comment:
Yeah. Do we mind about this?

Reply:
Does feel a bit funny. I wrote an [`AsyncKeyedLock`](https://github.com/dlcs/protagonist/blob/master/DLCS.Core/Threading/AsyncKeyedLock.cs) that handles locking in async code. 

Would mean changing this to return `Task<T>` which seems reasonable if it's making an async call.

Reply:
If this is a big 'zombie plague' moment for changing all calls, and this is working okay, we could open a ticket to fix?

Reply:
I think it's a good time to fix with the amount ported so far. There will be lots more code coming across that uses this so getting it the way we want it now and then fixing up the incoming stuff as it comes is (I think) a better path. 

I'll see if I can adapt this to use your AsyncKeyedLock.

Reply:
I've added it and triggered a cascade... seems to have settled down and still working.